tests/utils/host_utils.cpp

```diff
@@ -421,7 +421,15 @@ void initComms(int, char **, int *const commDims)
     QMP_declare_logical_topology_map(commDims, 4, map, 4);
   }
 #elif defined(MPI_COMMS)
-  MPI_Init(&argc, &argv);
+  int provided = 0;
+  int required = MPI_THREAD_FUNNELED;
+  int flag = MPI_Init_thread(&argc, &argv, required, &provided);
+
+  if (provided != required) {
```

Comment:
Is there a specific reason why we want this to error out if it's not the case? Can we do a cleanup and re-initialize with vanilla `MPI_Init` if we can't get funneled?

Reply:
Every MPI supports this now, and if we use `MPI_Init` we are skating on thin ice of UB given that QUDA is now threaded on the CPU.  So erroring out is the correct thing to do here.